base/math.jl

```diff
@@ -293,7 +293,7 @@ end
 
 # polynomial evaluation using compensated summation.
 # much more accurate, especially when lo can be combined with other rounding errors
-@inline function exthorner(x, p::Tuple)
+Base.@assume_effects :total  @inline function exthorner(x::Union{Float16, Float32, Float64}, p::Tuple)
```

Comment:
Can we get away with weaker effects here (just :terminates) by replacing the tuple getindex by getfield?

Reply:
yes, but imo, that's not noticably better.

Reply:
yes it is. `:total` is a very strong annotation that's easy to accidentally invalidate and cause segfaults.

Reply:
#46134

Reply:
hmm,
```
Base.@assume_effects :terminates_globally @noinline g(p::Tuple) 
    s=0
    for i in length(p):-1:1
        s+=getfield(p,i)
    end
    s
end
Base.infer_effects(g, (NTuple{5,Int},))
(!c,+e,!n,+t,+s)
```
Any idea what's going wrong here?

Reply:
Do you have #46044 merged?

Reply:
good point.